public/app/features/explore/RawListContainer.tsx

```diff
@@ -0,0 +1,110 @@
+import { css } from '@emotion/css';
+import { cloneDeep } from 'lodash';
+import React from 'react';
+import { VariableSizeList as List } from 'react-window';
+
+import { DataFrame, formattedValueToString } from '@grafana/data/src';
+import { stylesFactory } from '@grafana/ui/src';
+
+import RawList from './RawList';
+
+export type instantQueryRawVirtualizedListData = { Value: string; __name__: string; [index: string]: string };
+
+export interface RawListContainerProps {
+  tableResult: DataFrame;
+}
+
+type instantQueryMetricList = { [index: string]: { [index: string]: instantQueryRawVirtualizedListData } };
+
+const getRawListContainerStyles = stylesFactory(() => {
+  return {
+    wrapper: css`
+      height: 100%;
+      overflow: scroll;
+    `,
+  };
+});
+
+/**
+ * transform dataFrame to instantQueryRawVirtualizedListData
+ * @param dataFrame
+ */
+const getListItemsFromDataFrameNew = (dataFrame: DataFrame): instantQueryRawVirtualizedListData[] => {
```

Comment:
This looks a bit too PromQL specific and doesn't seem to work with for example MySQL. If "Raw" table view is a switcher on the table it should be generic to work with any tabular data:


https://user-images.githubusercontent.com/745532/192708624-e4b9f56b-e7cd-4749-885e-d542ba3e1408.mov

Reply:
@ifrost I'm not sure this view makes sense for MYSQL as it does for prometheus, MYSQL is always going to be tabular, and there's no string representation of a MYSQL row that can also be used as a query.

Reply:
Oh yeah fair enough, my point was mainly that the table visualization is generic and it can be used by results coming from any data source (for example MySQL), so the "raw" button would show up on other cases as well and it shouldn't show blank results, it looks confusing.

Reply:
Yeah good point, I'll take a look at getting that fixed now!

Reply:
I just pushed a quick change to only show this UI for prometheus